### PR TITLE
IDEM-245

### PR DESCRIPTION
**Pull request by pondpiu**
#245 
Verify otp screen

Image from iOS
Image from iOS (2)
Image from iOS (1)

Waiting for proxy endpoint to be implemented

### CODE REVIEW

**Review by bitcoinbrisbane — COMMENTED** [review]
Looks good, just need a few screen shots @pondpiu

File: app/src/navigation/screens/ClaimScreen.tsx [hunk 1]
```diff
@@ -219,17 +216,36 @@ const ClaimScreen: React.FC = () => {
...
-        <Button
-          title={isVerifying ? "Save & Verify" : "Save"}
-          disabled={!canSave}
-          onPress={onSave}
-          loading={loading}
-        />
+        {isOtpVerifyAction ? (
+          <Button
+            title={"Verify"}
+            disabled={!canSave}
```
Comment (brettvniekerk) [inline app/src/navigation/screens/ClaimScreen.tsx:238]: So is it disabled if the user canSave? Double negative hard to understand

**Review by brettvniekerk — APPROVED** [review]

**Comment by bitcoinbrisbane** [comment]
Need to address the 2 sonarqube issues too

File: app/src/navigation/screens/ClaimScreen.tsx (hunk 1 above)
Comment (brettvniekerk) [inline app/src/navigation/screens/ClaimScreen.tsx:238]: So is it disabled if the user canSave? Double negative hard to understand

File: app/src/navigation/screens/ClaimScreen.tsx [hunk 2]
```diff
@@ -219,17 +216,36 @@ const ClaimScreen: React.FC = () => {
...
+            onPress={openVerifyOtpScreen}
+          />
+        ) : (
+          <Button
+            title={isVerifying ? "Save & Verify" : "Save"}
+            disabled={!canSave}
```
Comment (brettvniekerk) [inline app/src/navigation/screens/ClaimScreen.tsx:244]: Same here

File: app/src/navigation/screens/ClaimScreen.tsx (hunk 1 above)
Comment (brettvniekerk) [inline app/src/navigation/screens/ClaimScreen.tsx:238]: So is it disabled if the user canSave? Double negative hard to understand

Comment (brettvniekerk) [inline app/src/navigation/screens/ClaimScreen.tsx:238]: So is it disabled if the user canSave? Double negative hard to understand

File: app/src/navigation/screens/ClaimScreen.tsx (hunk 2 above)
Comment (brettvniekerk) [inline app/src/navigation/screens/ClaimScreen.tsx:244]: Same here

**Review by brettvniekerk — CHANGES_REQUESTED** [review]

**Comment by xiangzhe0812** [comment]
@pondpiu removed verify otp screen, instead using Alert.prompt to input verification code

File: app/src/context/ClaimsStore.tsx [hunk 3]
```diff
@@ -7,7 +7,12 @@ import { displayClaimValue } from "../utils/claim-utils";
 export type ClaimsVault = {
   unclaimedClaims: Claim[];
   usersClaims: ClaimWithValue[];
-  addClaim: (claimId: ClaimType, value: any, files: string[]) => Promise<void>;
+  addClaim: (
+    claimId: ClaimType,
+    value: any,
```
Comment (brettvniekerk) [inline app/src/context/ClaimsStore.tsx:12]: Can you avoid any type?

**Review by brettvniekerk — DISMISSED** [review]

**Review by bitcoinbrisbane — DISMISSED** [review]

**Review by xiangzhe0812 — DISMISSED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

**Review by xiangzhe0812 — APPROVED** [review]

**Pull request merged**